awx/main/scheduler/dag_simple.py

```diff
@@ -28,97 +60,166 @@ def __iter__(self):
         return self.nodes.__iter__()
 
     def generate_graphviz_plot(self):
-        def short_string_obj(obj):
-            if type(obj) == Job:
-                type_str = "Job"
-            if type(obj) == AdHocCommand:
-                type_str = "AdHocCommand"
-            elif type(obj) == InventoryUpdate:
-                type_str = "Inventory"
-            elif type(obj) == ProjectUpdate:
-                type_str = "Project"
-            elif type(obj) == WorkflowJob:
-                type_str = "Workflow"
-            else:
-                type_str = "Unknown"
-            type_str += "%s" % str(obj.id)
-            return type_str
+        def run_status(obj):
+            dnr = "RUN"
+            status = "NA"
+            if hasattr(obj, 'job') and obj.job and hasattr(obj.job, 'status'):
+                status = obj.job.status
+            if hasattr(obj, 'do_not_run') and obj.do_not_run is True:
+                dnr = "DNR"
+            return "{}_{}_{}".format(dnr, status, obj.id)
 
         doc = """
         digraph g {
         rankdir = LR
         """
         for n in self.nodes:
+            obj = n['node_object']
+            status = "NA"
+            if hasattr(obj, 'job') and obj.job:
```

Comment:
If `job` is non-null, then this results in `O(N)` queries. See comment in DAG workflow file.

Also, consider that the `job` you get here isn't the same as the task being processed by the task manager, because a job could have changed state between when the task manager starts, and when it loads the nodes. That may or may not be a major problem.

Reply:
oh, b/c `generate_graphviz_plot`. Yep.